File: 6KYU/Change your Points of View/solution.clj
(ns running-lambda.core)

(defn point [a b]
   (fn [] [a b]))
  
(defn fst [point]
  (first (point)))
  
(defn snd [point]
  (second (point)))
  
(defn sqr-dist [point1 point2]
  (let [delta-x (Math/abs (- (fst point2) (fst point1)))
        delta-y (Math/abs (- (snd point2) (snd point1)))]
        (+ (* delta-x delta-x) (* delta-y delta-y))))
        
(defn line [foo bar]
  [(- (snd foo) (snd bar))
   (- (fst bar) (fst foo))
   (- (* (fst foo) (snd bar))
         (* (snd foo) (fst bar)))])

File: 6KYU/Change your Points of View/solution.cs
using System;

class PointView
{
	public static Func<int[]> Point(int a, int b)
	{
		  return () => new[]{a, b};
	}
	
  public static int Fst(Func<int[]> pt)
	{
		  return pt()[0];
	}
	
  public static int Snd(Func<int[]> pt)
	{
		  return pt()[1];
	}
	
  public static int SqrDist(Func<int[]> pt1, Func<int[]> pt2)
	{
    	var dx = Math.Abs(Fst(pt2) - Fst(pt1));
      var dy = Math.Abs(Snd(pt2) - Snd(pt1));
      return (dx * dx) + (dy * dy);
	}
  
	public static int[] Line(Func<int[]> pt1, Func<int[]> pt2) 
	{
	    return new[]
      {
         Snd(pt1) - Snd(pt2),
         Fst(pt2) - Fst(pt1),
         (Fst(pt1) * Snd(pt2)) - (Snd(pt1) * Fst(pt2))
      };
	}
}